src/ShellCheck/Analytics.hs

```diff
@@ -393,6 +431,18 @@ replaceToken id params r =
 surroundWith id params s = fixWith [replaceStart id params 0 s, replaceEnd id params 0 s]
 fixWith fixes = newFix { fixReplacements = fixes }
 
+-- Make a map from functions/aliases to definition IDs
+analyse f t = execState (doAnalysis f t) []
+functions t = Map.fromList $ analyse findFunctions t
+findFunctions (T_Function id _ _ name _) = modify ((name, id):)
+findFunctions t@(T_SimpleCommand id _ (_:args))
+    | t `isUnqualifiedCommand` "alias" = mapM_ getAlias args
+findFunctions _ = return ()
+getAlias arg =
+    let string = onlyLiteralString arg
+    in when ('=' `elem` string) $
+        modify ((takeWhile (/= '=') string, getId arg):)
+
```

Comment:
This snippet was moved from here:

https://github.com/koalaman/shellcheck/blob/da7b28213ef65ef4a0d702831d1d358f37e4f114/src/ShellCheck/Analytics.hs#L2244-L2252

The only difference is that `functions` was made to take an argument `t`.